regolith/mongoclient.py

```diff
@@ -125,6 +128,54 @@ def load_mongo_col(col: Collection) -> dict:
     }
 
 
+def doc_cleanup(doc: dict):
+    doc = bson_cleanup(doc)
+    doc['_id'].replace('.', '')
+    return doc
+
+
+def bson_cleanup(doc: dict):
+    """
+    This method should be used prior to updating or adding a document to a collection in mongo. Specifically, this
+    replaces all periods in keys and _id value with a blank, and changes datetime.date to an iso string. It does so
+    recursively for nested dictionaries.
+
+    Parameters
+    ----------
+    doc
+
+    Returns
+    -------
+    doc
+
+    """
+    def change_keys_id_and_date(obj, convert):
+        """
+        Recursively goes through the dictionary obj and replaces keys with the convert function.
+        """
+        if isinstance(obj, datetime.date):
+            # Mongo cannot handle datetime.date format, but we have infrastructure to handle iso date strings present
+            # Do not convert to datetime.datetime. Mongo can handle this, but regolith cannot.
+            return obj.isoformat()
```

Comment:
this means we can only support py >= 3.7 or 3.8 I think?  It is ok, but when we release it we need to make it clear which versions of python we are supporting.   There is extensive use of fstrings so we shouldn't support anything older than 3.6 or whenever they came in, for sure.  I think it is ok for it to be >= 3.# or wherever isoformat came in.  The code is not widely used outside the group at this point I think.  That could change in the future as we develop features and user interfaces, but it does mean that we can now insist on cetain versions of Python and make use of their features....there is no large group of legacy users.....

Reply:
https://stackoverflow.com/questions/3401428/how-to-get-current-isoformat-datetime-string-including-the-default-timezone

looks like isoformat() is use-able in 3.2+? Not sure how to check more thoroughly than with stack overflow tbh.

I am putting a python version restriction into setup.py now for fstrings.